firebase_admin/__init__.py

```diff
@@ -186,3 +197,41 @@ def credential(self):
     @property
     def options(self):
         return self._options
+
+    def get_token(self):
+        """Returns an OAuth2 bearer token.
+
+           This method may return a cached token. But it handles cache invalidation, and therefore
+           is guaranteed to always return unexpired tokens.
+
+           Returns:
+             string: An unexpired OAuth2 token.
+        """
+        if not self._token_valid():
+            self._token = self._credential.get_access_token()
+        return self._token.access_token
+
+    def _token_valid(self):
+        if self._token is None:
+            return False
+        skewed_expiry = self._token.expiry - datetime.timedelta(seconds=_CLOCK_SKEW_SECONDS)
+        return _clock() < skewed_expiry
+
+    def _get_service(self, name, initializer):
```

Comment:
A service is something we expose from the SDK (e.g. auth, database). By attaching each service to an `App` instance, we can gracefully terminate/stop them when the `App` is deleted. Here's the equivalent implementation from Java SDK: https://github.com/firebase/firebase-admin-java/blob/master/src/main/java/com/google/firebase/internal/FirebaseService.java

I've added docstrings to further explain this in the source.